jefferson/cli.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+import sys
+from pathlib import Path
+import contextlib
+import os
+import mmap
+
+import click
+
+from jefferson.cli_options import verbosity_option
+from jefferson.core.jffs2 import extract_jffs2
+
+@click.command(
+    help="A tool to extract JFFS2 filesystems.", context_settings=dict(help_option_names=["--help", "-h"])
+)
+@click.argument(
+    "file",
+    type=click.Path(path_type=Path, dir_okay=False, exists=True, resolve_path=True),
+    required=True,
+)
+@click.option(
+    "-d",
+    "--dest",
+    type=click.Path(path_type=Path, dir_okay=True, file_okay=False, resolve_path=True),
+    default=Path.cwd().joinpath("jffs2-root"),
+    help="Extract the files to this directory. Will be created if doesn't exist.",
+)
+@click.option(
+    "-f",
+    "--force",
+    is_flag=True,
+    show_default=True,
+    help="Force extraction even if outputs already exist (they are removed).",
+)
+@verbosity_option
+def cli(
+    file: Path,
+    dest: Path,
+    force: bool,
+    verbose: int,
+) -> int:
+    
+    if dest.exists() and not force:
+        print("Destination path already exists!")
+        return -1
+    dest.mkdir(exist_ok=True)
+    return extract_jffs2(file, dest, verbose)
+
+
+def main():
```

Comment:
The old argparse code is still left. I would delete this CLI interface and include it in the next PR with a version bump where we can clean the old code too, or you can delete that in the same commit.

Reply:
Deleted the old code within the commit that introduce click. Good catch, I didn't see I left it there.